### DISCUSSION TITLE
New Idea, System Corruption Checker, System Corruption Deletion tool, and scd /decorrupt command for Powershell

### DISCUSSION THREAD

**Discussion by EdgeDev01** [PowerShell/PowerShell#20708] [Enhancements]
the idea is scd /decorrupt this would add scd aka System Corruption Deletion tool, and that tool will be updated and used to check and remove corruption and replace every system file by downloading replacements from the internet and will remove every program the user added

**Comment by rhubarb-geek-nz** [comment]
On Windows, what would this do that [Use the System File Checker tool to repair missing or corrupted system files](https://support.microsoft.com/en-au/topic/use-the-system-file-checker-tool-to-repair-missing-or-corrupted-system-files-79aa86cb-ca52-166a-92a3-966e85d4094e) does not already do?

Why remove every program a user added? Adding a program to a computer is not corrupting a computer, it is a completely legitimate activity.

What you are describing is not decorruption, but attempting to return a computer to initial operating system installation state, which can already be done with [Give your PC a Fresh Start](https://support.microsoft.com/en-us/windows/give-your-pc-a-fresh-start-0ef73740-b927-549b-b7c9-e6f2b48d275e)

**Comment by EdgeDev01** [comment]
well i have a pc that cannot be given a fresh start no matter i try the system itself is corrupt

**Reply by rhubarb-geek-nz** [reply]
If the existing tools cannot recover the system then what would the tool you describe do differently that would fix the problem?

**Comment by EdgeDev01** [comment]
it also downloads dll files from the internet from reliable sources (but not without checking the file for malware and inconsistency first) and removes suspected corrupt dlls and replaces them with the ones that look identical and have no malware (that the scan saw)

**Reply by rhubarb-geek-nz** [reply]
This is what the existing tools _already_ do.

> When you run this command, DISM uses Windows Update to provide the files that are required to fix corruptions.

**Comment by EdgeDev01** [comment]
and by the way the command prompt running the tools would be in the background, accessing these running them as a background process as admin, however the powershell script would show a progress bar in this format (======100.0%======) and next to it wuld say what step for example 0.00% through 24.99% is initializing the process, 25.00% through 49.99% is running systemfix (all system fix tools, such as all the dism commands avalible, all chkdsk commands, and all sfc commands).

**Comment by EdgeDev01** [comment]
then 50.00% upto 74.99%, was all running netsh advfirewall reset, netsh int ip reset, and netsh winsock reset, and 75.00% upto 100.0% is just replacing corrupt dlls if found any or any suspected dlls that wern't found by any system tools

**Comment by EdgeDev01** [comment]
it has a 98.76% chance of fixing the problem however if it doesn't fix the problem it could be a faulty pc or install, or you landed on 2.34% where its corrupted hardware or hardware-level malware causing the problem

**Comment by EdgeDev01** [comment]
like if the program is a trojan and greyed the uninstall button and windows defender can't remove it the tool will

**Comment by EdgeDev01** [comment]
also the background process the powershell prompt created after running this command could be terminated

**Comment by EdgeDev01** [comment]
my version also checks earlier versions of windows, restore points, and microsoft update catelog, internet archive amoung others to revert the dll files as needed.

**Reply by rhubarb-geek-nz** [reply]
Ah, I misunderstood, so this is a tool that you have already written? Did it fix the PC you said could not be restored?
